pydantic/_internal/_core_utils.py

```diff
@@ -189,13 +204,13 @@ def _build_schema_type_to_method(self) -> dict[core_schema.CoreSchemaType, Recur
         return mapping
 
     def walk(self, schema: core_schema.CoreSchema, f: Walk) -> core_schema.CoreSchema:
-        return f(schema.copy(), self._walk)
+        return f(schema, self._walk)
```

Comment:
Does this have behavioural changes (e.g. walking schema now mutates inplace rather than copying)?

Reply:
Not really, I think we don’t mutate in place anyway